ignition-server/controllers/tokensecret_controller.go

```diff
@@ -284,11 +285,17 @@ func (r *TokenSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		r.PayloadStore.Set(string(oldToken), CacheValue{Payload: payload, SecretName: tokenSecret.Name})
 	}
 
-	// Store the cached payload in the secret to be consumed by in place upgrades.
 	patch := tokenSecret.DeepCopy()
-	patch.Data[TokenSecretPayloadKey] = payload
-	patch.Data[TokenSecretReasonKey] = []byte(hyperv1.NodePoolAsExpectedConditionReason)
-	patch.Data[TokenSecretMessageKey] = []byte("Payload generated successfully")
+	nodePoolUpgradeType, ok := tokenSecret.Annotations[TokenSecretNodePoolUpgradeType]
+	if ok && nodePoolUpgradeType == string(hyperv1.UpgradeTypeReplace) {
```

Comment:
nit: the ok && is redundant here, personally I find easier to read
```
	patch.Data[TokenSecretPayloadKey] = payload
	nodePoolUpgradeType := tokenSecret.Annotations[TokenSecretNodePoolUpgradeTyp"]
	if nodePoolUpgradeType == string(hyperv1.UpgradeTypeReplace) {
		delete(patch.Data, TokenSecretPayloadKey)
	}
```
just a subject opinion fwiw, no need to change.

Reply:
Can you please add a testCase here https://github.com/openshift/hypershift/blob/main/ignition-server/controllers/tokensecret_controller_test.go#L47